### DISCUSSION TITLE
LegacyCoordinator

### DISCUSSION THREAD

**Discussion by junbio** [mozilla-mobile/firefox-ios#11348] [General]
I was looking at the technical roadmap and saw this item: https://github.com/mozilla-mobile/firefox-ios/projects/7#card-64997151 

>Wrap existing app inside of a LegacyCoordinator

Could someone please explain what this means? Thanks!

**Comment by junbio** [comment]
@lmarceau Sorry to bother you, but I hope you aren't too busy to answer my question. Thanks!

**Reply by lmarceau** [reply]
Hello @junbio, this roadmap hasn't been updated in a while it seems, so I cannot really say what this task actually means. But I'll tag @electricRGB here since I know he's currently thinking on how we could better communicate our current roadmap 😄

**Comment by adudenamedruby** [comment]
Oh boy. Yeah, that is WOEFULLY out of date, unfortunately. THat is a holdover from when we were using GitHub as our main tool to track stuff. However, we've moved to JIRA and have left that floundering. Hopefully, I can get to updating it soon... although, I'm not sure if that's the best thing to keep track of things in a public way, seeing as then we're just doing double bookeeping... which is less than ideal.

One of my goals is to figure out a way to communicate the things the team is working on, on a quarterly, basis, to the community at large. That way, the community knows what parts of the code are safe to contribute to, and what's happening/what they can expect. Because our priorities are a mix of tech debt, product, and design, the the roadmap will reflect that and help give a better idea of the state of the project.

**Comment by junbio** [comment]
Thanks for the response. I was looking for what the LegacyCoordinator meant but I searched online for coordinator pattern and I think I get the general idea now. Most of the items on the roadmap, though outdated, are understandable such as "Refactor AppDelegate to use SceneDelegate".

**Reply by lmarceau** [reply]
It indeed seems that at that moment (some years ago) the team wanted to use the coordinator pattern. The current team thought is defining the direction we wanna take and currently coordinators are probably not going to be the answer. Items on that technical roadmap are out of date and I wouldn't look at it tbh.
